String.cxx

```diff
@@ -0,0 +1,130 @@
+}
+
+[[nodiscard]] inline bool endsWith(const String& s) const noexcept {
+    if (s.m_u.size() > m_u.size())
+        return false;
+    return (m_u.compare(m_u.length() - s.m_u.length(), s.m_u.length(), s.m_u) == 0);
+}
+
+inline void getBytes(byte buffer[], unsigned length) {
+    for (int i = 0; i < m_u.length; i++) {
+        buffer[i] = (byte)m_u[i];
+    }
+}
+
+[[nodiscard]] inline int indexOf(const char* c) const noexcept { return m_u.find(c); }
+
+[[nodiscard]] inline int indexOf(const char* c, unsigned index) const noexcept { return m_u.find(c, index); }
+
+[[nodiscard]] inline int indexOf(const String& str) const noexcept { return m_u.find(str); }
+
+[[nodiscard]] inline int indexOf(const String& str, unsigned index) const noexcept { return m_u.find(str, index); }
+
+inline void remove(unsigned idx) { m_u.erase(index); }
+
+inline void remove(unsigned idx, unsigned count) { m_u.erase(idx, idx + count - 1); }
+
+inline void replace(const String& substring1, const String& substring2) {
+    size_t position = m_u.find(substring1.m_u);
+
+    while (position != std::string::npos) {
+        m_u.replace(position, m_u.size(), substring2.m_u);
+        position = m_u.find(substring1.m_u, position + substring2.m_u.size());
+    }
+}
+
+inline void reserve(unsigned size) { m_u.reserve(size); }
+
+inline void setCharAt(unsigned index, const char* c) { m_u[index] = c; }
+
+[[nodiscard]] inline String substring(unsigned from) const { return m_u.substr(from); }
+
+[[nodiscard]] inline String substring(unsigned from, unsigned to) const { return m_u.substr(from, to - from); }
+
+inline void toCharArray(const char* buffer[], unsigned length) noexcept {
+    if (length <= m_u.length) {
+        for (int i = 0; i < length; i++) {
+            buffer[i] = m_u[i];
+        }
+    } else {
+        for (i = 0; i < m_u.length; i++) {
+            buffer[i] = m_u[i];
+        }
+    }
+}
+
+[[nodiscard]] inline long toInt() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stoi(m_u);
+        } else return 0;
+}
+
+[[nodiscard]] inline double toDouble() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stod(m_u);
+        } else return 0;
+}
+
+[[nodiscard]] inline float toFloat() const noexcept {
+        if (isdigit(m_u[0]) {
+        return std::stof(m_u);
+        } else return 0;
+}
+
+inline void toLowerCase() { std::transform(m_u.begin(), m_u.end(), m_u.begin(), std::tolower); }
+
+inline void toUpperCase() { std::transform(m_u.begin(), m_u.end(), m_u.begin(), std::toupper); }
+
+inline void trim() { m_u = m_u.erase(std::remove(m_u.begin(), m_u.end(), ' '), m_u.end()); }
+
+[[nodiscard]] inline bool equals(const String& s) const noexcept { return m_u == s.m_u; }
+
+[[nodiscard]] inline bool equalsIgnoreCase(const String& s) const noexcept {
+    String str = m_u;
+    return (std::transform(s.begin(), s.end(), s.begin(), std::tolower) == std::transform(str.begin(), str.end(), str.begin(), std::tolower));
+}
```

Comment:
These are not out-of-line definitions, but they should be
Prefix the function name by `classname::` to fix, and remove all `inline` specifiers from functions defined out of line
Example: 
```cpp
[[nodiscard]] bool String::equals(const String& s) const noexcept {
    return m_u == s.m_u;
}
```